src/components/widgets/Widget.tsx

```diff
@@ -24,7 +24,7 @@ export const Widget = ({ title, subtitle, children }: Props) => {
           {subtitle && <h2 className="font-normal">{subtitle}</h2>}
         </div>
         <div className="flex justify-end w-[60px]">
-          <button onClick={() => goBack()}>
+          <button onClick={() => handleExit()}>
```

Comment:
redundant arrow function.
should just be <button onClick={handleExit}>
and handleExit should useCallback.

And even better - force parent to pass goBack and simply use it